addons/stock/views/stock_picking_views.xml

```diff
@@ -286,8 +286,8 @@
                             </div>
                             <field name="partner_id" nolabel="1"/>
                             <field name="picking_type_id" attrs="{'invisible': [('hide_picking_type', '=', True)], 'readonly': [('state', '!=', 'draft')]}"/>
-                            <field name="location_id" options="{'no_create': True}" groups="stock.group_stock_multi_locations" attrs="{'invisible': [('picking_type_code', '=', 'incoming')]}"/>
-                            <field name="location_dest_id" options="{'no_create': True}" groups="stock.group_stock_multi_locations" attrs="{'invisible': [('picking_type_code', '=', 'outgoing')]}"/>
+                            <field name="location_id" options="{'no_create': True}" domain="[('usage', '!=', 'view')]" groups="stock.group_stock_multi_locations" attrs="{'invisible': [('picking_type_code', '=', 'incoming')]}"/>
```

Comment:
Remove this and change the location_id/location_dest_id to add this domain.
Change _get_putaway_locations to never return a view